ffmpeg/ffmpeg.go

```diff
@@ -171,6 +172,28 @@ func Transcode3(input *TranscodeOptionsIn, ps []TranscodeOptions) (*TranscodeRes
 	return t.Transcode(input, ps)
 }
 
+// Force copy-transcoding for audio-only segments
+func BypassTranscoding(input *TranscodeOptionsIn, ps []TranscodeOptions) (*TranscodeResults, error) {
+	glog.Errorf("WARNING: Input %s has audio & video streams but no video frames. Bypassing the transcoder.\n", input.Fname)
+	for i, p := range ps {
+		if p.Profile != (VideoProfile{}) {
+			p.Profile = (VideoProfile{})
+			if "drop" != p.VideoEncoder.Name && "copy" != p.VideoEncoder.Name {
+				p.VideoEncoder = ComponentOptions{Name: "copy"}
+			}
+			if "drop" != p.AudioEncoder.Name && "copy" != p.AudioEncoder.Name {
+				p.AudioEncoder = ComponentOptions{Name: "copy"}
+			}
+			ps[i] = p
+		}
+	}
+	// use a new thread to transcode
+	t := NewTranscoder()
```

Comment:
What is the advantage of creating a new transcoder in this function as opposed to just overriding the video profiles to use the copy video & audio encoder in `Transcode()`? 

i.e. in `Transcode()`

```golang
if !t.started {
    ret := int(C.lpms_is_bypass_needed(C.CString(input.Fname)))
    if ret != 1 {
        t.started = true
    } else {
        // Override video profiles to use copy video & audio encoder
        // Then, proceed with the rest of the Transcode() method
    }
}
```

Reply:
Yeah I tried that at first.

The problem happens on a subsequent normal segment in the same stream/session. We initialize the decoder/encoder contexts at the start of the session, so the Cgo module is unable to initialize a `non-copy` segment in a session that was started with a  `copy` segment and fails with [an error here](https://github.com/livepeer/lpms/blob/da8d5c9b1d0fd06db086a102f1281137255c288d/ffmpeg/lpms_ffmpeg.c#L652) while opening the stream. I avoided messing around with that logic for this.